### PR TITLE
Fix #13123, Only use products total in OrderTotalRuleChecker to select shipping methods

### LINKED ISSUE #13123: Incoherent total price in OrderTotalRuleChecker
**Sylius version affected**: 1.8, 1.9, 1.0

**Description**  
The rules `OrderTotalGreaterThanOrEqualRuleChecker` and `OrderTotalGreaterThanOrEqualRuleChecker` use the order total to check if a shipping method is available. Depending on when the rule is called, the total can be different for the same cart. For example:

* When computing the list of shipping method for the checkout form the price is the complete total (products + shipping + taxes + promotions).
* During the order processing steps, the price is the total without the adjustements (only products without taxes) since it runs just after OrderAdjustementsClearer.

It results in incoherent shipping method throughout the checkout.

**Steps to reproduce**  
1. Create one shipping method A available if total is bellow 100 with a price of 10.
1. Create a second shipping method B available if total is above 100.
1. Create a cart where items total is bellow 100 but above 91, default shipping method should therefore be A.
1. On the shipping method selection form, only shipping method B is available (items total + estimated shipping cost of A > 100). Only A should be available in this form.

**Possible Solution**  
User `$order->getItemsTotal()` instead of `$order->getTotal()`.

I'm not sure if only products should contribute to the shipping method rule checker. Maybe taxes and promotion should contribute too. But at least shipping method cost should not contribute to shipping method rule checker.

Maybe also update the priority of the steps of order processors to recompute products price and taxes before selecting the shipping method as mentioned in https://github.com/Sylius/Sylius/issues/12554.

### PR DESCRIPTION
**Pull request by mdevlamynck**
Only use products total in OrderTotalRuleChecker to select shipping methods. Not sure if it should be a breaking change or not.

| Q               | A
| --------------- | -----
| Branch?         | 1.9
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no?
| Deprecations?   | no
| Related tickets | fixes #13123
| License         | MIT

### CODE REVIEW

**Comment by Zales0123** [comment]
Hello Matthias and thank you very much for this fix! One more question before we can merge it. Do you think you would be able to construct a Behat scenario reproducing this bug? We would then be sure that it's red before and green after this change :) You've written so detailed reproduction steps in the [related issue](https://github.com/Sylius/Sylius/issues/13123), that they almost seem like a scenario 😄 🚀

**Comment by mdevlamynck** [comment]
@Zales0123 Done :)

**Comment by Snowbaha** [comment]
Hello, when do you plan to merge it?

**Review by Snowbaha — APPROVED** [review]
I confirm the fix!

**Comment by Snowbaha** [comment]
Hello, Any news from the fix? Why it is not merge yet?

**Comment by stloyd** [comment]
@lchrusciel @Zales0123 This seems correct, please check it.

**Comment by CoderMaggie** [comment]
Replaced by #13870

**Pull request closed without merging**
